### PR TITLE
use the same function semantics for table functions as for aggregates…

### PR DESCRIPTION
**Pull request by dobe**
… and scalars

### CODE REVIEW

File: sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java [hunk 1]
```diff
@@ -465,15 +462,16 @@ public Symbol resolveField(QualifiedName qualifiedName, @Nullable List path, Ope
             null
         );
 
-        List<Symbol> arguments = new ArrayList<>(node.arguments().size());
-        for (Expression expression : node.arguments()) {
-            Symbol symbol = expressionAnalyzer.convert(expression, context.expressionAnalysisContext());
-            arguments.add(symbol);
+        Function function = (Function) expressionAnalyzer.convert(node.functionCall(), context.expressionAnalysisContext());
+        FunctionImplementation functionImplementation = functions.getSafe(function.info().ident());
+        if (functionImplementation.info().type() != FunctionInfo.Type.TABLE){
```
Comment (matriv) [inline sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java]: space before {

File: sql/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java [hunk 2]
```diff
@@ -1910,19 +1910,12 @@ public void testSelectStarFromUnnest() throws Exception {
...
     public void testSelectStarFromUnnestWithInvalidArguments() throws Exception {
-        expectedException.expect(IllegalArgumentException.class);
-        expectedException.expectMessage("unnest expects arguments of type array. Got an argument of type 'long' at position 0 instead.");
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("unknown function: unnest(long, string)");
         analyze("select * from unnest(1, 'foo')");
     }
 
     @Test
-    public void testSelectStarFromUnnestWithNoArguments() throws Exception {
```
Comment (matriv) [inline sql/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java]: Why did you remove this test?

File: sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java [hunk 3]
```diff
@@ -465,15 +462,16 @@ public Symbol resolveField(QualifiedName qualifiedName, @Nullable List path, Ope
...
+            throw new UnsupportedFeatureException(
+                "non table function " + function.info().ident().name() + " is not supported in from clause");
```
Comment (matriv) [inline sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java]: start message with capital?

File: sql/src/test/groovy/io/crate/integrationtests/TableFunctionITest.groovy [hunk 4]
```diff
@@ -88,4 +89,18 @@ class TableFunctionITest extends SQLTransportIntegrationTest {
         execute("select col1 from unnest([1, 2]) where col1 = 2")
         assert printedTable(response.rows()) == "2\n"
     }
+
+    @Test
+    public void testSelectFromTableFunctionInSelectList() throws Exception {
+        expectedException.expect(SQLActionException.class)
+        expectedException.expectMessage(" Table functions are not supported in select list")
```
Comment (matriv) [inline sql/src/test/groovy/io/crate/integrationtests/TableFunctionITest.groovy]: space before Table

File: sql/src/test/groovy/io/crate/integrationtests/TableFunctionITest.groovy [hunk 5]
```diff
@@ -88,4 +89,18 @@ class TableFunctionITest extends SQLTransportIntegrationTest {
         execute("select col1 from unnest([1, 2]) where col1 = 2")
         assert printedTable(response.rows()) == "2\n"
     }
+
```
Comment (matriv) [inline sql/src/test/groovy/io/crate/integrationtests/TableFunctionITest.groovy]: I would move these tests to SelectStatementAnalyzerTest

File: sql/src/main/java/io/crate/operation/tablefunctions/Unnest.java [hunk 6]
```diff
@@ -26,84 +26,134 @@
...
+            final String localNodeId = clusterService.localNode().getId();
+            return new StaticTableInfo(TABLE_IDENT, columnRegistrar, Collections.emptyList()) {
+                @Override
+                public RowGranularity rowGranularity() {
+                    return RowGranularity.DOC;
+                }
+
+                @Override
+                public Routing getRouting(WhereClause whereClause, @Nullable String preference) {
+                    // XDOBE: check if we need the local nodeid and this table
```
Comment (matriv) [inline sql/src/main/java/io/crate/operation/tablefunctions/Unnest.java]: Remove comment?

File: sql/src/main/java/io/crate/operation/AbstractFunctionModule.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        resolver.put(name, functionResolver);
+    }
+
+    public abstract void configureFunctions();
+
+    @Override
+    protected void configure() {
+        configureFunctions();
+        // bind all registered functions and resolver
+        // by doing it here instead of the register functions, plugins can also use the
```
Comment (matriv) [inline sql/src/main/java/io/crate/operation/AbstractFunctionModule.java]: I don' get this phrase: "instead of the register functions"

File: sql/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java (hunk 2 above)
Comment (dobe) [inline sql/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java]: because this case is now covered also by the other argument validation test, since it is no special case anymore to have no arguments

File: sql/src/main/java/io/crate/operation/AbstractFunctionModule.java (hunk 7 above)
Comment (dobe) [inline sql/src/main/java/io/crate/operation/AbstractFunctionModule.java]: i think this is a historic thing (it was not changed with this PR) i will remove it

File: sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java (hunk 1 above)
Comment (dobe) [inline sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java]: hm somehow this is not covered by our editorconfig
